### DISCUSSION TITLE
How to improve the performance if i am scrolling speedily.

### DISCUSSION THREAD

**Discussion by sakshimorya90** [generic-ui/generic-ui#39] [General]
How to improve the performance , because when i am scrolling top to bottom so my CPU usage is around 50% to 90%. So please give me a solution for this. i need CPU usage below 40% if am scrolling speedily.

**Comment by generic-ui** [comment]
Have you tried https://generic-ui.com/guide/virtual-scroll? What is the size of your data, how many items & columns?

**Reply by sakshimorya90** [reply]
Yes, i am using Virtual scrolling and data size is unlimited (like infinite means 5 lakhs) and but we need to show in desktop at 13 item and mobile 8 item at a time.

**Reply by generic-ui** [reply]
So does virtual scroll meets your requirements? It should handle 500 000 items, there is an example for one million https://generic-ui.com/examples/big-data.
What kind of views are you using? So views may be more complicated to render and may decrease the performance.

**Reply by sakshimorya90** [reply]
Yes, i checked this link many times, but i also see there CPU process is using around 80% to 90% when i am scrolling from top to bottom continuously. So there is any other way to decrease the render time below to 40% to 50%.
![Screenshot from 2021-10-14 13-32-08](https://user-images.githubusercontent.com/92502396/137276348-5f03d035-86f9-4fe8-8826-9bbf131f40cf.png)
I have shared screen of CPU processing. so please check i need this process is below from 40-50%.

**Reply by generic-ui** [reply]
🤔  Atm I guess there is no way to decrease the CPU usage. May I ask why it is important to you?

**Reply by sakshimorya90** [reply]
Because we need to implement with some big gaming project , so there is another process is running that why we are need this process below down from 50%.

my requirement 
1. CPU < 40-50% (around 30%)
2. Js heap size not > 30mb
3. DOM Nodes -  2000

**Reply by sakshimorya90** [reply]
So i am going to try another lib those are bet for my game like ..vue.js ,angular js and core js.

if you got any solution so please let me know.
i will try again. 
Thanks

**Reply by generic-ui** [reply]
Very interesting requirements, I will think about it. Thanks for letting us know about your project. We constantly work on the performance as it is one of the most important features. Please revisit out project in the future.

Best luck with your gaming project.
